Source/Mock.cs

```diff
@@ -406,6 +406,7 @@ private static bool AreSameMethod(Expression left, Expression right)
 		private static string FormatSetupsInfo(IEnumerable<IProxyCall> setups)
 		{
 			var expressionSetups = setups
+				.Reverse()
```

Comment:
This is needed for continuity (so setups are shown in exception messages in the order in which they were registered). Minor performance penalty here, but since it only happens in the error case, that's OK.